app/models/iiif_canvas_indexer_enqueuer.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+##
+# For a given object in an exhibit, gets the correct manifest, and enqueues
+# canvas/page indexing jobs from that manifest.
+class IiifCanvasIndexerEnqueuer
+  attr_reader :exhibit, :druid
+
+  ANNOTATION_LIST = 'sc:AnnotationList'.freeze
+
+  def initialize(exhibit, druid)
+    @exhibit = exhibit
+    @druid = druid
+  end
+
+  def enqueue_jobs
+    canvases.each do |canvas|
+      canvas.other_content.each do |other_content|
+        # Don't bother to index unless there is an annotationList content to do so.
+        next unless other_content['@type'] == ANNOTATION_LIST
+        IndexCanvasJob.perform_later(other_content['@id'], canvas.to_json, exhibit)
```

Comment:
Recording something that came up in a group review of this: there may not necessarily be an @id  or @type. Manifests may only list `otherContent` as a plain string (no object or @id). Not a problem for us now, but we may want to ask about it.